docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,88 @@
+
+## List of Expanded Information 
+
+#### Compiler Version
+
+This will be a full compiler version, and if possible the commit SHA for the build of the compiler. This will be used since compilations accros compiler versions are not guaranteed to be the same.
+
+#### Signing Key
+
+Only public keys will be included. The following command line switches trigger a signing key: 
+
+* `-keyfile`
+* `-keycontainer`
+
+Open question: Do we need to know if `-delaysign` or `-publicsign` was specified? 
+
+#### Source Encoding
+
+Encoding for source files that are not declared using BOM. Do we need this?
+
+#### Path Map
+
+Required to replicate any path 
+
+#### Resources
+
+How do we handle `-linkresource`?
+
+`-resource` embeds resources into the output assembly. The order of the resources in the output file is determined from the order specified on the command line. Resource files have a `filename`, `identifier`, and `accessibility-modifier` for each resource. `filename` is the only required property, `identifier` defaults to `filename` if not specified, and `accessibility-modifier` defaults to public. 
+
+#### Metadata References
+
+Symbol server uses a [key](https://github.com/dotnet/symstore/blob/master/docs/specs/SSQP_Key_Conventions.md#pe-timestamp-filesize) computed from the COFF header in the PE image: 
+
+Timestamp: 4 byte integer
+Size of image: 4 byte integer
+
+Example:
+
+File name: `Foo.exe`
+
+COFF header Timestamp field: `0x542d5742`
+
+COFF header SizeOfImage field: `0x32000`
+
+Lookup key: `foo.exe/542d574232000/foo.exe`
+
+To fully support metadata references, we'll need to be able to find the exact PE image that was used in the compilation. We'll do this by storing the parts that make up the symbol server key.
+
+#### Platform
+
+
+
+## PDB Format Additions
+
+#### Metadata References
+
+Metadata references can be easily stored as binary. The binary encoding will be as follows: 
+
+Timestamp Field: 4 byte integer
+Size of field: 4 byte integer
+Name: A UTF-16 little-endian encoded string following the PrimitiveConstant values in [Local Constants](https://github.com/jnm2/dotnet-runtime/blob/26efe3467741fe2a85780b2d2cd18875af6ebd98/docs/design/specs/PortablePdb-Metadata.md#localconstant-table-0x34)
+
+??: Does name need to be null-terminated? Does size need to be included? Is there a max size? 
+
+#### Key Value Pairs
+
+The remaining values will be stored as key value pairs in the pdb. The storage format will be UTF8 encoded JSON. These are the known keys that will be used: 
+
+1. "compilerversion" : string
+2. "signingkey" : string
+3. "sourceencoding" : string
+4. "pathmap": string
+5. "platform": string
```

Comment:
What's platform?

Reply:
anycpu, x86, arm64, etc. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/platform-compiler-option 

I wasn't 100% sure about this, but it seemed it might be necessary in case the compilation includes variances based on platform?

Reply:
Oh, I see. These flags can be read from the PE file.

Reply:
Checkout: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Emit/ModulePropertiesForSerialization.cs,a1f03a0489115277
This struct is passed to the emitter and I believe all of it can be reconstructed from the PE headers.

Reply:
Here's how the struct maps back to compilation options: http://sourceroslyn.io/#Microsoft.CodeAnalysis/Compilation/Compilation.cs,1774

Reply:
We'd definitely want tests that validate that we can reconstruct it.

Reply:
Essentially, you can start by a test that creates CSharpCompilationOptions and EmitOptions and tries to fill in all the values. Each value can either be found in the PE headers/metadata or we need to store it in the PDB.

Reply:
See PEHeaders: https://source.dot.net/#System.Reflection.Metadata/System/Reflection/PortableExecutable/PEHeaders.cs,df912b397b528b66

Reply:
We will need these:

```
-checked[+|-]                 Generate overflow checks
-unsafe[+|-]                  Allow 'unsafe' code
-define:<symbol list>         Define conditional compilation symbol(s) (Short
                              form: -d)
-langversion:?                Display the allowed values for language version
-langversion:<string>         Specify language version such as
                              `latest` (latest version, including minor versions),
                              `default` (same as `latest`),
                              `latestmajor` (latest version, excluding minor versions),
                              `preview` (latest version, including features in unsupported preview),
                              or specific versions like `6` or `7.1`
-nullable[+|-]                Specify nullable context option enable|disable.
-nullable:{enable|disable|warnings|annotations}
                              Specify nullable context option enable|disable|warnings|annotations.
```

Reply:
All of these are now covered